unlock-js/src/__tests__/integration/walletServiceIntegration.test.js

```diff
@@ -138,7 +139,111 @@ describe('Wallet Service Integration', () => {
+          owner = accounts[5]
+
+          if (lock.currencyContractAddress === null) {
+            // Get the ether balance of the lock before the purchase
+            lockBalanceBefore = await web3Service.getAddressBalance(lockAddress)
+            // Get the ether balance of the user before the purchase
+            userBalanceBefore = await web3Service.getAddressBalance(sender)
+          } else {
+            // Get the erc20 balance of the lock before the purchase
+            lockBalanceBefore = await getErc20BalanceForAddress(
+              lock.currencyContractAddress,
+              lockAddress
+            )
+            // Get the erc20 balance of the user before the purchase
+            userBalanceBefore = await getErc20BalanceForAddress(
+              lock.currencyContractAddress,
+              sender
+            )
+          }
+
+          tokenId = await walletService.purchaseKey({
+            lockAddress,
+            owner,
+            keyPrice: lock.keyPrice,
+          })
+          key = await web3Service.getKeyByLockForOwner(lockAddress, owner)
+        })
+
+        it('should yield the tokenId', () => {
+          expect.assertions(1)
+          expect(tokenId).not.toBe(null) // We don't know very much beyond the fact that it is not null
+        })
+
+        it('should have increased the currency balance on the lock', async () => {
+          expect.assertions(1)
+          let newBalance
+          if (lock.currencyContractAddress === null) {
+            newBalance = await web3Service.getAddressBalance(lockAddress)
+          } else {
+            newBalance = await getErc20BalanceForAddress(
+              lock.currencyContractAddress,
+              lockAddress
+            )
+          }
+          expect(parseFloat(newBalance)).toEqual(
+            parseFloat(lockBalanceBefore) + parseFloat(lock.keyPrice)
+          )
+        })
+
+        it('should have decreased the currency balance of the person making the purchase', async () => {
+          expect.assertions(1)
+          let newBalance
+          if (lock.currencyContractAddress === null) {
+            newBalance = await web3Service.getAddressBalance(sender)
+          } else {
+            newBalance = await getErc20BalanceForAddress(
+              lock.currencyContractAddress,
+              sender
+            )
+          }
+
+          if (lock.currencyContractAddress === null) {
+            // For Ether we need to account for gas
+            expect(parseFloat(newBalance)).toBeLessThan(
+              parseFloat(userBalanceBefore) - parseFloat(lock.keyPrice)
+            )
+          } else {
+            // For ERC20 the balance should be exact
+            expect(parseFloat(newBalance)).toBe(
+              parseFloat(userBalanceBefore) - parseFloat(lock.keyPrice)
+            )
+          }
+        })
+
+        it('should have assigned the key to the right user', async () => {
```

Comment:
I think you have the descriptions for this test and the following one swapped

Reply:
oups.

Reply:
fixed!